#ver=2.0
---
# The controller_prop are used to define the behaviors of the controller.
# The variables under controller_prop are related to the ControllerInfo.msg.
controller_prop:
    msg_type : 0 # Set to MSG_TYPE_STEERING_WHEEL or MSG_TYPE_CHASSIS. See ControllerInfo.msg for more details.
    service_mode : 0 # Set to SERVICE_MODE_CLIENT or SERVICE_MODE_SERVER. See ControllerInfo.msg for more details.
    service_name : "controller" # The _id number will added in suffix, i.e. "<service_name>_<id>".
    timeout_ms : 30.0 # Tell control server to wait for response for _ms.
    period_ms : 50.0 # Tell control server to send control signal every _ms.
    privilege : 100 # Privileges. Range from 0 to 100.
    pub_type : 0 # Set to PUB_TYPE_XXX. See ControllerInfo.msg for more details.

input_prop:
    mode : "joystick" # Set to "joystick" or "keyboard".
    joystickConfigFile : "controller_joystick.json" # The joystick configuration file. Ignored if mode is "keyboard".
    keyboardConfigFile : "controller_keyboard.json" # The keyboard configuration file. Ignored if mode is "joystick".

# Settings for Params class under vehicle_interfaces/params.h
generic_prop:
    # Node name and namespace. The namespace will automatically adding prefix at node name, topic name, service name, etc..
    namespace : "V0"
    nodeName : "controller"

    # Device ID, the ID will automatically formed the name of nodes and publisher topics. For instance, 
    # the node name will be formed as "<nodeName>_<id>_node", and the publisher topic name will be formed as "<topic_name>_<id>". 
    # Note: If the topic use to be subscriptions, then the topic name will not be formed.
    id : 0

    # ################################################################################################
    # The rest of params will not be used. Change the following params under service.json instead.
    # ################################################################################################

    # Device information service configuration
    devInfoService : "" # Service name devinfo_0
    devInterface : "eth0" # eth0
    devMultiNode : True # False

    # QoS service configuration
    qosService : "qos_0" # Service name qos_0
    qosDirPath : "launch/qos"

    # Safety service configuration
    safetyService : "" # Service name safety_0

    # Time sync service configuration
    timesyncService : "timesync_0" # Service name timesync_0
    timesyncPeriod_ms : 600000.0 # 10 minute a sync for HPC. 2 minute a sync for sensors.
    timesyncAccuracy_ms : 2.0 # Restrict the duration of sync process time
    timesyncWaitService : False # Wait until time synced at construction.